### PR TITLE
refactor(input format): refactor with FieldEncoder.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

part of https://github.com/datafuselabs/databend/issues/7732

### CODE REVIEW

File: tests/logictest/suites/base/03_dml/03_0023_insert_into_array [hunk 1]
```diff
@@ -389,11 +389,11 @@ ee ff
 statement ok
 CREATE TABLE IF NOT EXISTS t18(id Int, arr Array(Int64 Null)) Engine = Fuse;
 
-statement ok
+statement error 1010
```
Comment (BohuTANG) [inline tests/logictest/suites/base/03_dml/03_0023_insert_into_array:392]: From `ok` to `1010` errors need comments here.

Comment (youngsofun) [reply tests/logictest/suites/base/03_dml/03_0023_insert_into_array:392]: ok,  I`ll do it latter:

``` ERROR 1105 (HY000): Code: 1010, displayText = Cast error happens in casting from Array(Nullable(UInt8)) to Array(Nullable(Int64)).``

Comment (youngsofun) [reply tests/logictest/suites/base/03_dml/03_0023_insert_into_array:393]: done

**Comment by youngsofun** [comment]
not related to this pr

```
--- /workspace/tests/suites/0_stateless/13_tpch/13_0009_q9.result	2022-11-14 05:45:08.409161832 +0000
+++ /workspace/tests/suites/0_stateless/13_tpch/13_0009_q9.stdout	2022-11-14 05:48:52.606210366 +0000
@@ -1,5 +1 @@
-ALGERIA	1998	2321785.368
-ALGERIA	1997	3685016.858
-ALGERIA	1996	4276597.425
-ALGERIA	1995	4418370.415
-ALGERIA	1994	3864849.952
+ERROR 1105 (HY000) at line 2: result row write failed: Code: 1067, displayText = error reading a body from connection: broken pipe, source: Some(hyper::Error(Body, Error { kind: Io(Kind(BrokenPipe)) })).
```

**Review by b41sh — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

**Comment by youngsofun** [comment]
not related to this pr 

```
Runner: http
ErrorType: statement query execute with exception
Message: Failed to execute. Collected info: 1006: Unable to get field named "0". Valid fields: []

Parsed Statement
    at_line: 401,
    s_type: Statement: query, type: I, query_type: I, retry: False,
    suite_name: query/join.test,
    text:
        select * from (SELECT number AS a FROM numbers(10)) x right join (SELECT number AS a FROM numbers(5))  y using(a) order by x.a;
    results: [(<re.Match object; span=(0, 4), match='----'>, 409, '0\n1\n2\n3\n4')],
    runs_on: {'http', 'clickhouse', 'mysql'},
```

**Comment by youngsofun** [comment]
> not related to this pr， and the error exists before the pr (also occur in other`s pr)
> 
> ```
> Runner: http
> ErrorType: statement query execute with exception
> Message: Failed to execute. Collected info: 1006: Unable to get field named "0". Valid fields: []
> 
> Parsed Statement
>     at_line: 401,
>     s_type: Statement: query, type: I, query_type: I, retry: False,
>     suite_name: query/join.test,
>     text:
>         select * from (SELECT number AS a FROM numbers(10)) x right join (SELECT number AS a FROM numbers(5))  y using(a) order by x.a;
>     results: [(<re.Match object; span=(0, 4), match='----'>, 409, '0\n1\n2\n3\n4')],
>     runs_on: {'http', 'clickhouse', 'mysql'},
> ```

can we merge this pr mannualy

**Pull request merged**